flink-libraries/flink-table/src/main/scala/org/apache/flink/table/codegen/CodeGenerator.scala

```diff
@@ -1562,6 +1561,39 @@ class CodeGenerator(
   }
 
   /**
+    * Adds a reusable constructor statement with the given parameter types,
+    * and the member variable's name that stores the variables passed in the constructor.
+    *
+    * @param memberVariables The member variable's name that stores the variables passed in
+    *                        the constructor.
+    * @param parameterTypes The parameter types to construct the function
+    */
+  def addReusableConstructorWithMemberVariables(
```

Comment:
This method is basically doing the same as `addReusableConstructor` except that it allows to hardcode the names of the member variables. The same functionality (in a safe manner) can be achieved by reusing existing code, IMO.

For instance, the code of the test could be rewritten to:

```
val params = generator.addReusableConstructor(classOf[Long])
    val card = params(0)

    val body =
      s"""
         |return java.lang.Long.valueOf(in1) + java.lang.Long.valueOf($card);
         """.stripMargin
```

We would have to remove the `null` assignment for the initially empty field variable in `addReusableConstructor` though (or check if it is a primitive type).